### PR TITLE
refactor: imports

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [X] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
No, Already present.

### Motivation / Use-Case

Reduce import at startup. 

### Breaking Changes
None

### Additional Info
No

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -246,6 +243,8 @@ class Server {
       // It is possible to use the `bypass` method without a `target`.
       // However, the proxy middleware has no use in this case, and will fail to instantiate.
       if (proxyConfig.target) {
+        const { createProxyMiddleware } = require('http-proxy-middleware');
```
Comment (alexander-akait) [inline lib/Server.js:246]: Move it here https://github.com/webpack/webpack-dev-server/blob/d8ffe3927b235e542eae23bab6e9dfdbea8e5bf0/lib/Server.js#L173 better we still will use it due https://github.com/webpack/webpack-dev-server/blob/d8ffe3927b235e542eae23bab6e9dfdbea8e5bf0/lib/Server.js#L421

Comment (snitin315) [reply lib/Server.js:246]: On it.

File: lib/Server.js [hunk 2]
```diff
@@ -980,6 +985,9 @@ class Server {
   }
 
   serveMagicHtml(req, res, next) {
+    const getFilenameFromUrl =
+      require('webpack-dev-middleware/dist/utils/getFilenameFromUrl').default;
```
Comment (alexander-akait) [inline lib/Server.js:989]: I think we should add this method to public `webpack-dev-middleware` API here https://github.com/webpack/webpack-dev-middleware/blob/master/src/index.js#L78, in future we should avoid import non public API

Comment (snitin315) [reply lib/Server.js:989]: Makes sense, will do 👍🏻

**Review by anshumanv — APPROVED** [review]
👍

File: lib/Server.js [hunk 3]
```diff
@@ -2,36 +2,18 @@
...
 const path = require('path');
 const url = require('url');
-const http = require('http');
-const https = require('https');
 const fs = require('graceful-fs');
 const ipaddr = require('ipaddr.js');
 const internalIp = require('internal-ip');
 const killable = require('killable');
-const chokidar = require('chokidar');
 const express = require('express');
```
Comment (alexander-akait) [inline lib/Server.js:9]: I think in future we reduce them more, good right now

**Review by alexander-akait — APPROVED** [review]

File: lib/Server.js [hunk 4]
```diff
@@ -546,6 +544,9 @@ class Server {
   }
 
   createServer() {
+    const https = require('https');
+    const http = require('http');
```
Comment (ylemkimon) [inline lib/Server.js:548]: I'm not sure it'd make a difference, but I think we can move these below just like `require('spdy')`.

Comment (alexander-akait) [reply lib/Server.js:548]: Feel free to send PR

**Pull request merged**